packages/core-sdk/resolvers/token-resolvers/resolvers/uniswap/src/query/functions/getTokenComponents.ts

```diff
@@ -0,0 +1,106 @@
+import { BigInt } from "@web3api/wasm-as";
+import { Big } from "as-big";
+
+import {
+  env,
+  Ethereum_Connection,
+  Ethereum_Query,
+  Input_getTokenComponents,
+  Interface_Token,
+  Interface_TokenComponent,
+  QueryEnv,
+  Token_Query,
+  Token_TokenType,
+} from "../w3";
+
+function getPairTokenAddresses(pairAddress: string, connection: Ethereum_Connection): string[] {
+  // get token addresses
+  const token0AddressResult = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token0() external view returns (address)",
+    args: [],
+    connection: connection,
+  });
+  // if exception encountered, pair contract presumed not to exist
+  if (token0AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  const token1AddressResult = Ethereum_Query.callContractView({
+    address: pairAddress,
+    method: "function token1() external view returns (address)",
+    args: [],
+    connection: connection,
+  });
+  if (token1AddressResult.isErr) {
+    throw new Error("Invalid protocol token");
+  }
+  return [token0AddressResult.unwrap(), token1AddressResult.unwrap()];
+}
+
+export function getTokenComponents(input: Input_getTokenComponents): Interface_TokenComponent {
+  if (env == null) throw new Error("env is not set");
+  const connection = (env as QueryEnv).connection;
+
+  const token = Token_Query.getToken({
```

Comment:
We need to do error handling now since we are throwing error in case of invalid token